addons/mail/static/src/components/composer/composer.js

```diff
@@ -178,6 +178,17 @@ class Composer extends Component {
      * @private
      */
     async _postMessage() {
+        if (!this.composer.canPostMessage) {
+            if (this.composer.hasUploadingAttachment) {
+                this.env.services['notification'].notify({
+                    message: this.env._t("Please wait while the file is uploading."),
+                    sticky: true,
+                    title: this.env._t("Uploading error"),
+                    type: 'warning',
+                });
+            }
+            return;
+        }
```

Comment:
With this additional behaviour, I would add some extra information that the posting of message could be aborted in case the message cannot be posted, such as when a file is being uploaded.